llvm/lib/Analysis/TypeBasedAliasAnalysis.cpp

```diff
@@ -821,13 +821,15 @@ MDNode *AAMDNodes::extendToTBAA(MDNode *MD, ssize_t Len) {
 AAMDNodes AAMDNodes::adjustForAccess(unsigned AccessSize) {
   AAMDNodes New = *this;
   MDNode *M = New.TBAAStruct;
-  New.TBAAStruct = nullptr;
   if (M && M->getNumOperands() == 3 && M->getOperand(0) &&
```

Comment:
Isn't this too strict ? Typically the TBAAStruct have operands in multiple of 3 describing {offset, size, type }.

You can extract the 'first member' instead of matching the full struct. (Note: if we accept to track unions in more detail, this might get more complicated)

Reply:
Yep, I left this to here to keep the changes small, I'll soon share this one in the chain.